### PR TITLE
[WebRTC] Explicit decline message for webrtc p2p sessions.

### PR DESCRIPTION
**Pull request by roxanneskelly**
WebRTC P2P uses a different mechanism for managing p2p voice calls than Vivox, as WebRTC uses the existing messaging infrastructure to handle calls, declines, accepts, etc.  Vivox uses a vivox-supplied mechanism for p2p calls.

This required us to add explicit p2p call and p2p decline messaging.

Also, in this PR, are some linux-related code fixes and some renaming of items to be more clear.

### CODE REVIEW

File: indra/llwebrtc/llwebrtc.cpp [hunk 1]
```diff
@@ -516,13 +515,23 @@ void LLWebRTCImpl::updateDevices()
         char name[webrtc::kAdmMaxDeviceNameSize];
         char guid[webrtc::kAdmMaxGuidSize];
         mTuningDeviceModule->PlayoutDeviceName(index, name, guid);
+
+#if WEBRTC_LINUX
```
Comment (Nicky-D) [inline indra/llwebrtc/llwebrtc.cpp:519]: I suggest turning this either into a function ala
```
LLWebRTCVoiceDevice createWebRTCVoiceDevice( std::string name, std::string guid )
{
#if LL_LINUX
    if( guid.empty() )
        guid = name;
#endif

    return {name, guid };
```

and then call this in the two places. Or just be lazy and modify the ctor of LLWebRTCVoiceDevice

```
    LLWebRTCVoiceDevice(const std::string &display_name, const std::string &id) :
        mDisplayName(display_name),
        mID(id)
    {
        if( mID.empty() )
            mID = mDisplayName;
    };
```

in any case is my suggestion to get rid of:
- the strcpy which can easily lead to buffer overflows if one of the enums above changes
- the duplicate code of checking for an empty GUID

File: indra/newview/llimview.cpp [hunk 2]
```diff
@@ -2727,7 +2727,7 @@ LLIncomingCallDialog::LLIncomingCallDialog(const LLSD& payload) :
 
 void LLIncomingCallDialog::onLifetimeExpired()
```
Comment (Nicky-D) [inline indra/newview/llimview.cpp:2728]: This function looks quite bad in the webview. I suspect it has spaces and tabs mixed. Settling on one would be preferred

File: indra/llwebrtc/llwebrtc_impl.h [hunk 3]
```diff
@@ -35,6 +35,7 @@
 #define WEBRTC_POSIX 1
 #elif __linux__
```
Comment (Nicky-D) [inline indra/llwebrtc/llwebrtc_impl.h:36]: I suppose the LL-way is to check for LL_LINUX, LL_WINDOWS, LL_DARWIN. This makes grepping easier.

Which at the same time makes me curious why there is a need for WEBRTC_xxx when there is already such a thing as LL_xxx.

And another small nitpick
LL_**WINDOWS** <> LL_**WIN**
LL_**DARWIN** <> WEBRTC_**MAC** (you're my hero btw if you'd manage to get DARWIN changed to OSX or MAC)

File: indra/llwebrtc/llwebrtc.cpp (hunk 1 above)
Comment (roxanneskelly) [reply indra/llwebrtc/llwebrtc.cpp]: Done (the second option)

File: indra/newview/llimview.cpp (hunk 2 above)
Comment (roxanneskelly) [reply indra/newview/llimview.cpp:2728]: Should be better now

File: indra/llwebrtc/llwebrtc_impl.h (hunk 3 above)
Comment (roxanneskelly) [reply indra/llwebrtc/llwebrtc_impl.h:36]: I'm trying to use the WebRTC way of doing everything (thread model, memory, platform identifiers, etc.) in llwebrtc.  It's the 'interface' that wraps webrtc in a very simple wrapper for the viewer.  Ultimately, the goal is to move it out of the viewer into something downloaded by autobuild (or whatever a TPV dev uses) where I won't have access to viewer-related headers and build-system constants.

File: indra/llwebrtc/llwebrtc.cpp (hunk 1 above)
Comment (Nicky-D) [reply indra/llwebrtc/llwebrtc.cpp]: Thank you :)

File: indra/newview/llimview.cpp (hunk 2 above)
Comment (Nicky-D) [reply indra/newview/llimview.cpp:2728]: Yes, looks awesome now

File: indra/llwebrtc/llwebrtc_impl.h (hunk 3 above)
Comment (Nicky-D) [reply indra/llwebrtc/llwebrtc_impl.h:36]: Got it, that makes sense.
For grepability  I'd still prefer _WINDOWS and _DARWIN (even though I'd personally prefer _MAC)
But then  the suffix is the same across the sources.

Comment (roxanneskelly) [reply indra/llwebrtc/llwebrtc_impl.h:36]: The constants listed are used in the webrtc headers so I'm kinda stuck using WEBRTC_*, unfortunately.

**Review by marchcat — APPROVED** [review]

**Pull request merged**